Src/Application/Extensions.cs

```diff
@@ -19,13 +13,22 @@ public static class Extensions
     public static IServiceCollection RegisterApplicationServices(this IServiceCollection services,
         IConfiguration configuration)
     {
-        services.AddScoped<IScoreCalculatorService<int, int>, AgeScoreCalculatorService>();
-
-        services.AddScoped<IScoreCalculatorService<int, int>, JobExperienceScoreCalculatorService>();
-        services.AddScoped<IScoreCalculatorService<int, int>, MaritalStatusScoreCalculatorService>();
-        services
-            .AddScoped<IScoreCalculatorService<LanguageMarkDtoList, LanguageMarkDtoList>, LanguageScoreCalculator>();
-
+        services.AddScoped<AgeScoreStrategy> ();
+        services.AddScoped<JobExperienceScoreStrategy> (); 
+        services.AddScoped<LanguageScoreCalculator> ();
+        services.AddScoped<MaritalStatusScoreStrategy> ();
+        services.AddScoped<UniversityDegreeCalculator> ();
+        services.AddScoped<Application.Services.TotalScoreCalculator> ();
+        
+        services.AddScoped<List<IScoreStrategy>>(provider => new List<IScoreStrategy>
+        {
```

Comment:
Can't we just inject all strategies that we know implement IScoreStrategy using reflection or other things if it is possible? @MehranLabour